### ISSUE TITLE
Unkown Error

### ISSUE THREAD

**Issue by AdultGear** [skaut/skaut-google-drive-gallery#1987]
I am getting an unknown error in the plugin settings as well as front of the site, This was working fine too.
<img width="381" alt="Screenshot 2023-05-22 at 9 58 47 PM" src="https://github.com/skaut/skaut-google-drive-gallery/assets/133848451/1ba630d3-fc88-465c-8ab7-c02c73c126d1">

**Comment by marekdedic** [comment]
Hi, could you please enable debug logging for your WordPress site? That way the plugin gives more info... Thanks :)

**Comment by AdultGear** [comment]
Sorry for delay in replying... before you replied I revoked access, granted it back and it was all working again. If I see the error again ill turn on debugging and see what it gives.

**Comment by marekdedic** [comment]
Ok, thanks!

**Comment by mxstudios** [comment]
Same here except error comes back after a few hours.

**Comment by mxstudios** [comment]
@AdultGear did you get this working? Did it stay authenticated after you revoked and regranted? Mine still only works 2 hours at a time.

**Comment by matthewxmurphy** [comment]
Nah it keeps doing the same thing. Sorry for delay in response I was having email issues.

**Comment by AdultGear** [comment]
> @AdultGear did you get this working? Did it stay authenticated after you revoked and regranted? Mine still only works 2 hours at a time.

Unfortunately, I continue to encounter recurring issues that resurface unexpectedly. It's frustrating because I believe the problem has been resolved, only to receive feedback from a customer indicating that it's still not working as intended. Please accept my sincere apologies for the delayed response as well. I experienced difficulties with my email and encountered challenges with signing into github.

**Comment by marekdedic** [comment]
Hi,
is the google app private or public an  in testing or production?

**Comment by 8ToThePowerOfMol** [comment]
Same behaviour at our site too. The site is running on the most recent wordpress v6.2.2. I also tried to turn off all other plugins and the bug is still there.

**Comment by 8ToThePowerOfMol** [comment]
[2005](https://github.com/skaut/skaut-google-drive-gallery/issues/2005), and probably [2070](https://github.com/skaut/skaut-google-drive-gallery/issues/2070) are the same issues as this one.

**Comment by marekdedic** [comment]
Hi, could you please upgrade to 2.13.4, turn on [debug mode](https://wordpress.org/documentation/article/debugging-in-wordpress/) and send me the output? Thanks

**Comment by mohwald** [comment]
@marekdedic 
```
The Image and video gallery from Google Drive plugin has encountered an error. Error message:

refresh token must be passed in or set as part of setAccessToken

Stack trace:

#0 /home/vu009503/www/wp-content/plugins/skaut-google-drive-gallery/class-api-client.php(116): Sgdg\Vendor\Google\Client->fetchAccessTokenWithRefreshToken()
#1 /home/vu009503/www/wp-content/plugins/skaut-google-drive-gallery/class-api-client.php(134): Sgdg\API_Client::get_authorized_raw_client()
#2 /home/vu009503/www/wp-content/plugins/skaut-google-drive-gallery/class-api-client.php(152): Sgdg\API_Client::get_drive_client()
#3 /home/vu009503/www/wp-content/plugins/skaut-google-drive-gallery/admin/settings-pages/basic/root-selection/class-list-ajax-endpoint.php(73): Sgdg\API_Client::preamble()
#4 /home/vu009503/www/wp-content/plugins/skaut-google-drive-gallery/helpers/class-helpers.php(42): Sgdg\Admin\Settings_Pages\Basic\Root_Selection\List_Ajax_Endpoint::ajax_handler_body()
#5 /home/vu009503/www/wp-content/plugins/skaut-google-drive-gallery/admin/settings-pages/basic/root-selection/class-list-ajax-endpoint.php(53): Sgdg\Helpers::ajax_wrapper()
#6 /home/vu009503/www/wp-includes/class-wp-hook.php(310): Sgdg\Admin\Settings_Pages\Basic\Root_Selection\List_Ajax_Endpoint::handle_ajax()
#7 /home/vu009503/www/wp-includes/class-wp-hook.php(334): WP_Hook->apply_filters()
#8 /home/vu009503/www/wp-includes/plugin.php(517): WP_Hook->do_action()
#9 /home/vu009503/www/wp-admin/admin-ajax.php(188): do_action()
#10 {main}
```

I will try Revoke -> Grant to see whether anything changed.

**Comment by Honny1** [comment]
After enabling debug mode I got this traceback. 

php: 7.3
wordpress: 6.3
plugin: 2.13.5

```
The Image and video gallery from Google Drive plugin has encountered an error. Error message:

[{ "error": { "code": 400, "message": "Failed to parse batch request, error: 0 items. Received batch body: (0) bytes redacted", "errors": [ { "message": "Failed to parse batch request, error: 0 items. Received batch body: (0) bytes redacted", "domain": "global", "reason": "badRequest" } ], "status": "INVALID_ARGUMENT" } } ]

Stack trace:

#0 /home/users/hony8ucz/hony.8u.cz/web/wp-content/plugins/skaut-google-drive-gallery/vendor/google/apiclient/src/Http/REST.php(82): Sgdg\Vendor\Google\Http\REST::decodeHttpResponse()
#1 /home/users/hony8ucz/hony.8u.cz/web/wp-content/plugins/skaut-google-drive-gallery/vendor/google/apiclient/src/Task/Runner.php(153): Sgdg\Vendor\Google\Http\REST::doExecute()
#2 /home/users/hony8ucz/hony.8u.cz/web/wp-content/plugins/skaut-google-drive-gallery/vendor/google/apiclient/src/Http/REST.php(53): Sgdg\Vendor\Google\Task\Runner->run()
#3 /home/users/hony8ucz/hony.8u.cz/web/wp-content/plugins/skaut-google-drive-gallery/vendor/google/apiclient/src/Client.php(739): Sgdg\Vendor\Google\Http\REST::execute()
#4 /home/users/hony8ucz/hony.8u.cz/web/wp-content/plugins/skaut-google-drive-gallery/vendor/google/apiclient/src/Http/Batch.php(92): Sgdg\Vendor\Google\Client->execute()
#5 /home/users/hony8ucz/hony.8u.cz/web/wp-content/plugins/skaut-google-drive-gallery/class-api-client.php(291): Sgdg\Vendor\Google\Http\Batch->execute()
#6 /home/users/hony8ucz/hony.8u.cz/web/wp-content/plugins/skaut-google-drive-gallery/vendor/google/apiclient/src/Task/Runner.php(153): Sgdg\API_Client::Sgdg\{closure}()
#7 /home/users/hony8ucz/hony.8u.cz/web/wp-content/plugins/skaut-google-drive-gallery/class-api-client.php(306): Sgdg\Vendor\Google\Task\Runner->run()
#8 /home/users/hony8ucz/hony.8u.cz/web/wp-content/plugins/skaut-google-drive-gallery/admin/settings-pages/basic/root-selection/class-list-ajax-endpoint.php(116): Sgdg\API_Client::execute()
#9 /home/users/hony8ucz/hony.8u.cz/web/wp-content/plugins/skaut-google-drive-gallery/helpers/class-helpers.php(42): Sgdg\Admin\Settings_Pages\Basic\Root_Selection\List_Ajax_Endpoint::ajax_handler_body()
#10 /home/users/hony8ucz/hony.8u.cz/web/wp-content/plugins/skaut-google-drive-gallery/admin/settings-pages/basic/root-selection/class-list-ajax-endpoint.php(53): Sgdg\Helpers::ajax_wrapper()
#11 /home/users/hony8ucz/hony.8u.cz/web/wp-includes/class-wp-hook.php(310): Sgdg\Admin\Settings_Pages\Basic\Root_Selection\List_Ajax_Endpoint::handle_ajax()
#12 /home/users/hony8ucz/hony.8u.cz/web/wp-includes/class-wp-hook.php(334): WP_Hook->apply_filters()
#13 /home/users/hony8ucz/hony.8u.cz/web/wp-includes/plugin.php(517): WP_Hook->do_action()
#14 /home/users/hony8ucz/hony.8u.cz/web/wp-admin/admin-ajax.php(188): do_action()
#15 {main}
```

**Comment by marekdedic** [comment]
Hi @AdultGear and @mxstudios, if you are still encountering the error with "refresh token must be passed", this was fixed properly in 2.13.13, I recommend updating and re-authorizing the plugin with Google.
